### PR TITLE
Prom scraper source id can be log-cache compatible

### PR DESCRIPTION
**Pull request by ablease**
This is the first PR to address this issue https://github.com/rabbitmq/service-operator-experience/issues/47

Raise a templating error when attempting to set source id with greater than 48 characters.

### CODE REVIEW

File: jobs/rabbitmq-server/templates/prom_scraper_config.yml.erb [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+    raise 'prom_scraper source_id must be 48 characters or less'
+  end
+elsif p('rabbitmq-server.cluster_name') == "" then
+  source_id = 'rabbit@localhost'
+else
+ source_id = p('rabbitmq-server.cluster_name')
+end
+-%>
+source_id: <%= source_id %>
 instance_id: <%= if p('rabbitmq-server.create_swap_delete') == true then "'rabbit@#{spec.address}'" else "'rabbit@#{Digest::MD5.hexdigest(spec.ip)}'" end %>
```
Comment (coro) [inline jobs/rabbitmq-server/templates/prom_scraper_config.yml.erb:17]: Don't know if you plan to do it in a follow-on PR or not, but this is missing the instance_id change. It's probably good to lump both together in the same PR so both changes end up in the same BOSH release.

File: jobs/rabbitmq-server/templates/prom_scraper_detailed_config.yml.erb [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+    raise 'prom_scraper source_id must be 48 characters or less'
+  end
+elsif p('rabbitmq-server.cluster_name') == "" then
+  source_id = 'rabbit@localhost'
+else
+ source_id = p('rabbitmq-server.cluster_name')
+end
+-%>
+source_id: <%= source_id %>
 instance_id: <%= if p('rabbitmq-server.create_swap_delete') == true then "'rabbit@#{spec.address}'" else "'rabbit@#{Digest::MD5.hexdigest(spec.ip)}'" end %>
```
Comment (coro) [inline jobs/rabbitmq-server/templates/prom_scraper_detailed_config.yml.erb:17]: As above.

**Review by coro — COMMENTED** [review]
source_id changes look great. I think you should include the instance_id changes in the same PR as mentioned in my comments.

**Review by coro — APPROVED** [review]

**Comment by ablease** [comment]
Thanks Connor!

**Pull request merged**